internal/super/generate.go

```diff
@@ -122,157 +133,185 @@ func generateScript(
 	_ flowkit.Services,
 	state *flowkit.State,
 ) (result command.Result, err error) {
-	return generateNew(args, "script", generateFlags.Directory, logger, state)
+	generator := NewGenerator(DefaultCadenceDirectory, state, logger, false)
+	err = generator.Create(TemplateMap{ScriptType: []string{args[0]}})
+	return nil, err
 }
 
-func addCDCExtension(name string) string {
-	if strings.HasSuffix(name, ".cdc") {
-		return name
-	}
-	return fmt.Sprintf("%s.cdc", name)
-}
+// TemplateMap defines a map of template types to their specific names
+type TemplateMap map[string][]string
 
-func stripCDCExtension(name string) string {
-	return strings.TrimSuffix(name, filepath.Ext(name))
+type Generator struct {
+	Directory   string
```

Comment:
i believe we can make these private now since you got the constructor params workign :)